authenticate/authenticate.go

```diff
@@ -102,14 +102,15 @@ func New(opts config.Options) (*Authenticate, error) {
 		return nil, err
 	}
 
-	// shared state encoder setup
+	// shared secret is used to encrypt and sign data shared between services
 	sharedCipher, _ := cryptutil.NewAEADCipherFromBase64(opts.SharedKey)
-	signedEncoder, err := jws.NewHS256Signer([]byte(opts.SharedKey), opts.AuthenticateURL.Host)
+	decodedSharedSecret, _ := base64.StdEncoding.DecodeString(opts.SharedKey)
+	signedEncoder, err := jws.NewHS256Signer(decodedSharedSecret, opts.AuthenticateURL.Host)
 	if err != nil {
 		return nil, err
 	}
 
-	// private state encoder setup
+	// cookie secret is used encrypt, and sign authenticate service payloads
```

Comment:
you missed a `to`  `cookie secret is used to encrypt,...`

Reply:
yes thank you good catch